python/pyspark/ml/param/__init__.py

```diff
@@ -389,8 +391,8 @@ def copy(self, extra=None):
         if extra is None:
             extra = dict()
         that = copy.copy(self)
-        that._paramMap = self.extractParamMap(extra)
-        return that
+        that._paramMap = {}
```

Comment:
I think it's OK to leave this as is.  It's within the Params class, and there is no need to use copyValues, which is meant to handle different classes.